src/cascadia/TerminalApp/App.cpp

```diff
@@ -713,14 +714,32 @@ namespace winrt::TerminalApp::implementation
     void App::_UpdateTitle(std::shared_ptr<Tab> tab)
     {
         auto newTabTitle = tab->GetFocusedTitle();
+        const auto lastFocusedProfileOpt = tab->GetFocusedProfile();
+        const auto lastFocusedProfile = lastFocusedProfileOpt.value();
+        const auto* const matchingProfile = _settings->FindProfile(lastFocusedProfile);
+
+        auto tabTitle = matchingProfile->GetTabTitle();

```

Comment:
Can't this just be...

```suggestion
        if (!tableTitle.empty())
        {
             tabTitle = winrt::to_hstring(matchingProfile->GetTabTitle());
        }
```

and then you can de-duplicate the contents of the if/else below since they're both exactly the same except for this parameter?

Reply:
tabTitle would still need to be defined before the if?